app/AvailablePlugin/EmailAddressWidget/Controller/EmailAddressWidgetVerificationsController.php

```diff
@@ -0,0 +1,139 @@
+ * regarding copyright ownership.
+ *
+ * UCAID licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link          http://www.internet2.edu/comanage COmanage Project
+ * @package       registry
+ * @since         COmanage Registry v4.1.0
+ * @license       Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ */
+
+App::uses("StandardController", "Controller");
+
+class EmailAddressWidgetVerificationsController extends StandardController {
+  // Class name, used by Cake
+  public $name = "EmailAddressWidgetVerifications";
+
+  public $uses = array(
+    'EmailAddressWidget.CoEmailAddressWidget',
+    'EmailAddressWidget.EmailAddressWidgetVerification',
+    'CoMessageTemplate'
+  );
+
+  /**
+   * Determine the CO ID based on some attribute of the request.
+   * This method is intended to be overridden by model-specific controllers.
+   *
+   * @since  COmanage Registry v4.1.0
+   * @return Integer CO ID, or null if not implemented or not applicable.
+   * @throws InvalidArgumentException
+   */
+
+  protected function calculateImpliedCoId($data = null) {
+    if(!empty($this->request->params['pass'][0])) {
+      $rec = $this->EmailAddressWidgetVerification->getRecordToVerify($this->request->params['pass'][0]);
+      if(isset($rec['CoEmailAddressWidget']["CoDashboardWidget"]["CoDashboard"]["co_id"])) {
+        return $rec['CoEmailAddressWidget']["CoDashboardWidget"]["CoDashboard"]["co_id"];
+      }
+    }
+
+    // Or try the default behavior
+    return parent::calculateImpliedCoId();
+  }
+
+  /**
+   * Step 2 in adding an email address:
+   * Verify the token associated with the new address.
+   *
+   * @since  COmanage Registry v4.1.0
+   */
+  public function verify($token) {
+    if(empty($token)) {
+      $this->Api->restResultHeader(HttpStatusCodesEnum::HTTP_BAD_REQUEST, _txt('er.token'));
+      return false;
+    }
+
+    // Retrieve the Verification record and the configuration
+    $rec = $this->EmailAddressWidgetVerification->getRecordToVerify($token);
+    if(empty($rec)) {
+      $this->Api->restResultHeader(HttpStatusCodesEnum::HTTP_NOT_FOUND, _txt('er.token'));
+      return false;
+    }
+    // Check if the person owns the records and can proceed with the verification
+    $CoPerson = ClassRegistry::init('CoPerson');
+    $actorCoPersonId = $CoPerson->idForIdentifier($this->cur_co["Co"]["id"], $this->Session->read('Auth.User.username'));
```

Comment:
I'm pretty sure I commented on this before. Don't do this identifier map. Just read `Auth.User.co_person_id` from the session.